### DISCUSSION TITLE
How can I add custom fields for the checkout shipping method block?

### DISCUSSION THREAD

**Discussion by wppunk** [woocommerce/woocommerce#43847] [Checkout Flow]
Hello everyone. Can someone help me figure out how to add a few fields under my shipping method?
![CleanShot 2024-01-21 at 23 23 42@2x](https://github.com/woocommerce/woocommerce-blocks/assets/19273242/4345fd0a-2e36-42c9-acb7-28e5e0d89425)
I want to have two select2 fields when my shipping method is selected.

For the checkout shortcode, I use the `woocommerce_after_shipping_rate` filter and `woocommerce_form_field` functions on PHP.

**Comment by opr** [comment]
Hi @wppunk - This repository (WooCommerce Blocks) is no longer in use. I'll move your discussion to the WooCommerce monorepo.

To answer your question, you can use [Slot/Fill](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/slot-fills.md) to add a component in the [`ExperimentalOrderShippingPackages` slot.](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-slot-fills.md#experimentalordershippingpackages)
